src/DsaLocal.cc

```diff
@@ -533,6 +533,62 @@ void IntraBlockBuilder::visitLoadInst(LoadInst &LI) {
   }
 }
 
+/// OldVal := *Ptr
+/// if OldVal == Cmp then
+///    *Ptr := New
+/// return OldVal
```

Comment:
Should this return the other value, i.e., `if *Ptr == New then Old else New`?

Reply:
This is the fixed semantics:
```
/// OldVal := *Ptr
/// Success := false
/// if OldVal == Cmp then
///    Success := true
///    *Ptr := New
/// return {OldVal, Success}
```